src/Raven.Server/RavenServerStartup.cs

```diff
@@ -277,7 +277,8 @@ private static void CheckDatabaseShutdownAndThrowIfNeeded(RequestHandlerContext
                 return;
 
             var databaseShutdown = context.Database?.DatabaseShutdown ?? context.DatabaseContext?.DatabaseShutdown;
-            if (databaseShutdown is { IsCancellationRequested: true })
+            var databaseShutdownCompleted = context.Database?.DatabaseShutdownCompleted;
+            if (databaseShutdown is { IsCancellationRequested: true } || databaseShutdownCompleted is { IsSet: true})
```

Comment:
@arekpalinski I think this should be the other way around... because the `databaseShutdown` can be already disposed.
And yes, we miss the concept of `databaseShutdownCompleted` for the orchestrator.

Reply:
Fixed in https://github.com/ravendb/ravendb/pull/18202